models/DataObject/ClassDefinition/Data.php

```diff
@@ -76,33 +76,30 @@ abstract class Data implements DataObject\ClassDefinition\Data\TypeDeclarationSu
      * @var string[]
      */
     protected const FORBIDDEN_NAMES = [
-        'id', 'key', 'path', 'type', 'index', 'classname', 'creationdate', 'userowner', 'value', 'class', 'list',
-        'fullpath', 'childs', 'children', 'values', 'cachetag', 'cachetags', 'parent', 'published', 'valuefromparent',
-        'userpermissions', 'dependencies', 'modificationdate', 'usermodification', 'byid', 'bypath', 'data',
-        'versions', 'properties', 'permissions', 'permissionsforuser', 'childamount', 'apipluginbroker', 'resource',
-        'parentClass', 'definition', 'locked', 'language', 'omitmandatorycheck', 'idpath', 'object', 'fieldname',
-        'property', 'parentid', 'scheduledtasks', 'latestVersion', 'haschildren', 'siblings', 'hassiblings',
-        'childrenSortby', 'childrensortorder', 'versioncount', 'dirtylanguages', 'dirtyfields', 'classTitle',
+        'apipluginbroker', 'byid', 'bypath', 'cachetag', 'cachetags', 'childamount', 'children', 'childrensortby',
+        'childrensortorder', 'childs', 'class', 'classid', 'classname', 'classtitle', 'closestparentofclass',
+        'creationdate', 'dao', 'data', 'definition', 'dependencies', 'dirtyfields', 'dirtylanguages', 'fieldname',
+        'fullpath', 'haschildren', 'hassiblings', 'id', 'idpath', 'index', 'key', 'language', 'latestversion',
+        'lazyloadedfieldnames', 'list', 'locked', 'modificationdate', 'nextparentforinheritance',
+        'object', 'omitmandatorycheck', 'parent', 'parentclass', 'parentid', 'path', 'permissions',
+        'permissionsforuser', 'properties', 'property', 'published', 'relationdata', 'resource', 'scheduledtasks',
+        'siblings', 'type', 'usermodification', 'userowner', 'userpermissions', 'value', 'valueforfieldname',
+        'valuefromparent', 'values', 'versioncount', 'versions',
```

Comment:
Thank you again for your PRs

I've found some more cases that we missed, basically some of the getter methods in  https://github.com/pimcore/pimcore/tree/11.x/models/DataObject/AbstractObject.php

realpath, realFullPath, currentfullpath, types(plural which returns CONSTS, the singular one is already covered) ,hideunpublished,getinheritedValues(method starts as getGet), childpermissions

Reply:
@kingjia90 I added it: https://github.com/pimcore/pimcore/pull/17109/commits/09ce06b41ad2f118ee23251190f6d41f005fb969